gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -665,9 +679,12 @@ private void viewFile(ProjectFile file, ProjectFileViewerExtension viewerExtensi
         Node graphic = viewerExtension.getMenuGraphic(file);
         ProjectFileViewer viewer = viewerExtension.newViewer(file, getContent().getScene(), context);
         Tab tab = new MyTab(tabName, viewer);
+        viewerList.add(viewer);
         tab.setOnCloseRequest(event -> {
             if (!viewer.isClosable()) {
                 event.consume();
+            } else {
+                viewerList.remove(viewer);
```

Comment:
Is it possible that the dispose method doesn’t close the viewer for any reason? Maybe the remove method should be called in the onClosed event listener instead of onCloseRequest listener.